### DISCUSSION TITLE
pv loss due to temperature

### DISCUSSION THREAD

**Discussion by 115522** [pvlib/pvlib-python#1612] [Q&A] [answered]
hello Everyone , I am new at pvlib and I want know how can I calculate PV loss due to temperature with pvlib I only found how to calculate cell temperature , any maths formula will be appriciated

amany thanks in advance

**Comment by cwhanse** [comment]
By loss due to temperature, what do you mean? One can calculate different energy values by assuming different temperature profiles. It seems to me that regarding a difference in energy as a loss due to temperature requires choosing one temperature profile as a reference.

**Reply by 115522** [reply]
exactly @cwhanse  , I agree with you but the temperature profile choosen as a reference is temp = 25 °C at STC 

power_loss = Power(temp_ambiante) - Power(temp_STC)

**Comment by cwhanse** [comment, marked as answer]
If you are using ModelChain, run once with the ambient air temperature and save the total energy (mc.results.ac.sum())

Then create a copy of the weather file, insert

weather_copy['temp_cell'] = 25.

and remove ambient temperature

weather_copy.drop('temp_air')

Then run ModelChain again and you should get a different cell temperature and AC energy.

I haven't coded this.

**Reply by 115522** [reply]
I can't use ModelChain I have my owen data weather and PAN file , but here I am asking about temperature cell  the model of sandia coded in pvlib ,it hasn't any effect on power DC ??

**Reply by cwhanse** [reply]
DC power is adjusted to cell temperature in pvlib. If that's not happening in your code, please post a sample here, there must be a mistake or omission.

**Reply by 115522** [reply]
ok here is my code to calculate dc power  : 

      ` meteo = pd.read_csv('src/meteo/weather_data.csv')
           # golobal incidence
            poa_global = meteo['GlobInc']
            #temperature ambiante 
            temp_air = meteo['Temperature']
       
      # cellule temperature from pvlib : cell_temperature determination depends on exterior temperature and irradiation
      cell_temp = pvlib.pvsystem.temperature.pvsyst_cell(poa_global, temp_air)
      temperature_difference =  cell_temp - temp_air
       IL, I0, Rs, Rsh, nNsVth = pvlib.pvsystem.calcparams_pvsyst(effective_irradiance =poa_global, temp_cell = cell_temp,
                                                         alpha_sc = module_parameters['alpha_sc'],
                                                         gamma_ref = module_parameters['gamma_ref'],
                                                         mu_gamma = module_parameters['mu_gamma'],
                                                         I_L_ref = module_parameters['I_L_ref'],
                                                         I_o_ref = module_parameters['I_o_ref'],
                                                         R_sh_ref = module_parameters['R_sh_ref'],
                                                         R_sh_0 = module_parameters['R_sh_0'],
                                                         R_s = module_parameters['R_s'],
                                                         cells_in_series = module_parameters['cells_in_series'])
      #power dc
      dc = pvlib.pvsystem.singlediode(IL, I0, Rs, Rsh, nNsVth)
      yearly_dc = dc.p_mp.sum() / 1000.0
      print('annual dc = {} kWh/yr'.format(yearly_dc))`

**Reply by cwhanse** [reply]
The adjustment for cell temperature happens in the `calcparams_pvsyst` step. Each of the single diode equation parameter (IL, I0, Rs, Rsh, nNsVth) is adjusted to the irradiance (poa_global) and cell temperature specified in the inputs. If you examine the values of nNsVth, they should be linear in cell temperature, for example.

**Reply by 115522** [reply]
very nice ! , than power dc is a dataframe which contains value for each irradiance and cell_temp , than how can I determine the loss energy ??

**Reply by cwhanse** [reply]
If you want to estimate the energy that would be produced if cell temperature was always 25C, then run the same code as above, but instead of using pvlib.pvsystem.temperature.pvsyst_cell, set

cell_temp = 25.

**Reply by JoaNIAS** [reply]
Hello! 
Could you please clarify, why mostly in every example I've seen, that as effective_irradance is taken poa_global value. 

`pvlib.pvsystem.calcparams_pvsyst(effective_irradiance=poa_global, `...

I thought its not totally the same. 
In pvlib, effective irradiance refers to the irradiance reaching a module's cells, i.e., 'poa_global', adjusted for spectrum and reflection.
Is there any point to calculate effective_irradiance more precise, instead of taking poa_global value, and whether it will make a big difference to the PVsystem forecast output?

Thank you in advance!
